### PR TITLE
make .gitignore easy to maintain

### PR DESCRIPTION
**Pull request by xunpan**
don't mess up .gitignore
Let's keep it simple and easy to maintain

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,145 +1,23 @@
...
 
-# OSX leaves these everywhere on SMB shares
-._*
-
-# OSX trash
-.DS_Store
-
-# Eclipse files
-.classpath
-.project
```
Comment (gyliu513) [inline .gitignore]: Need keep .project as it will be generated by eclipse.

File: .gitignore [hunk 2]
```diff
@@ -1,145 +1,23 @@
-*.json
-*.sublime-project
-*.sublime-workspace
+# temporay files
 *.swp
-.idea
-.DS_Store
```
Comment (gyliu513) [inline .gitignore]: Keep .DS_Store for mac os.

File: .gitignore [hunk 3]
```diff
@@ -1,145 +1,23 @@
...
-.classpath
-.project
-.settings/**
-
-# Files generated by JetBrains IDEs, e.g. IntelliJ IDEA
-.idea/
-*.iml
-
-# Vscode files
-.vscode
```
Comment (gyliu513) [inline .gitignore]: Keep .vscode for VSCode.

File: .gitignore [hunk 4]
```diff
@@ -1,145 +1,23 @@
...
-# OSX trash
-.DS_Store
-
-# Eclipse files
-.classpath
-.project
-.settings/**
-
-# Files generated by JetBrains IDEs, e.g. IntelliJ IDEA
-.idea/
```
Comment (gyliu513) [inline .gitignore]: Keep this for IDEA.

**Comment by xunpan** [comment]
Actually, I'm thinking what this file is used for in best practise.

 - For project related files: generated, temprary files, etc. We need to ignore it for sure.
 - For editor/OS related files, do we need to keep it in `.gigignore` for all?
   At first, I guess so because it helps developers. https://github.com/kubernetes-sigs/kubefed/pull/871.
However, after thinking it over, I think developer specific files should better not in this file. It does not help all development process. Developer maintains their own environment seems better because they have their own editor preferences: https://help.github.com/en/articles/ignoring-files#create-a-global-gitignore.

Any comments?

**Comment by gyliu513** [comment]
Thanks @xunpan make sense.

/lgtm

**Pull request merged**